Server/sprawl_calculator/wup_calculator/views.py

```diff
@@ -1,13 +1,78 @@
 from django.shortcuts import render
 from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework import status
+from datetime import datetime
+from dataclasses import asdict
+from .models import WUPCalculation
+from django.http import JsonResponse
+from .data_definations import WUPCalculationResult, WUPCalculationRequestActivating, WUPCalculationRequestProcessing, WUPCalculationRequestCompleted, WUPCalculationRequestRejected, WUPCalculationRequestError
 # Create your views here.
 
-#Endpoint1_View
+
+""" Start a WUP index calculation processing request """
+
 @api_view(['PUT', 'GET'])
 def wup_index_generator(request):
-    raise NotImplementedError
+    if request.data.get('resident_count_in_boundary') is None or 
+        request.data.get('employment_count_in_boundary') is None or 
+        request.data.get('raster_with_build_up_area') is None or 
+        request.data.get('raster_no_data_value') is None or 
+        request.data.get('raster_build_up_value') is None or 
+        request.data.get('vector_boundary') is None:
+        return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
A comment, a alternate way would be to create a dataclass called `WUPIndexGeneratorRequest` and then populate the fields there so that we can be clear as to the request parameters.

Reply:
ok,will createdataclass  `'WUPIndexGeneratorRequest'`.let me that check how to check is valid or not.

Reply:
Thanks once you create the class and try to create a instence it will tell you if the data in the instance is valid. There is no special validation necessary.

Reply:
here i found [from_dict()](https://pypi.org/project/dict-to-dataclass/l) metho which convert dictionaries to instances of dataclasses.and asdict() converts instances into dictionary.

Reply:
That method converts from Dictionary -> Dataclass  what we need is a Dataclass -> Dictionary that method is available in the standard library